client/js/ifml/tojson/elementrules.js

```diff
@@ -0,0 +1,220 @@
+            return {
+                elements: {
+                    id: element.id,
+                    attributes: {
+                        collection: element.get('collection') || ''
+                    }
+                }
+            };
+        }
+
+    ),
+    createRule(
+        function (element, model) {
+            return model.isList(element);
+        },
+        function (element) {
+            return {
+                elements: {
+                    id: element.id,
+                    attributes: {
+                        filters: (element.get('filters') && element.get('filters').slice()) || []
+                    }
+                }
+            };
+        }
+
+    ),
+    createRule(
+        function (element, model) {
+            return model.isViewContainer(element);
+        },
+        function (element) {
+            return {
+                elements: {
+                    id: element.id,
+                    attributes: {
+                        name: element.get('name'),
+                        default: element.get('default'),
+                        landmark: element.get('landmark'),
+                        xor: element.get('xor')
+                    },
+                    metadata: {
+                        graphics: {
+                            size: element.get('size')
+                        }
+                    }
+                }
+            };
+        }
+    ),
+    createRule(
+        function (element, model) {
+            return model.isFlow(element);
+        },
+        function (element) {
+            var vertices = element.get('vertices'),
+                statistics = element.get('statistics');
+            return {
+                elements: {
+                    id: element.id,
+                    type: element.get('type'),
+                    attributes: {
+                        bindings: element.get('bindings').slice()
+                    },
+                    metadata: {
+                        graphics: {
+                            vertices: vertices && vertices.slice()
+                        },
+                        statistics: statistics && statistics.slice()
+                    }
+                }
+            };
+        }
+    ),
+    createRule(
+        function (element, model) {
+            return model.isChildElement(element);
+        },
+        function (element) {
+            return {
+                relations: {
+                    type: 'hierarchy',
+                    parent: element.get('parent'),
+                    child: element.id
+                }
+            };
+        }
+    ),
+    createRule(
+        function (element, model) {
+            return model.isFlow(element);
+        },
+        function (element) {
+            return {
+                relations: {
+                    type: 'source',
+                    flow: element.id,
+                    source: element.get('source').id
```

Comment:
It could be interesting to uniform the calls in the file to nested properties either using _get_ or _prop_.